### PR TITLE
Add logger in Submit Url for Scan action when domain is on blacklist in Urlscan plugin

### PR DESCRIPTION
**Pull request by r7-kszczepanskagorna**
## Proposed Changes
Add logger in Submit Url for Scan action when domain is on blacklist in Urlscan plugin

### Description
Add logger in Submit Url for Scan action when domain is on blacklist in Urlscan plugin and return empty 'scain_id' for this case. Ticket MC-411

Describe the proposed changes:

  -

### Testing

Any testing information goes here.

## PR Requirements

Developers, verify you have completed the following items by checking them off:

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [ ] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [ ] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [ ] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [ ] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [ ] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `icon-plugin run -c sample $action > tests/$action.json`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -R all --debug --jq` (use PR format at end)

### Run

<details>

```
{
  "body": {
    "log": "rapid7/urlscan.io:2.3.0. Step name: submit_url_for_scan\nThe submitted domain is on our blacklist. For your own safety we did not perform this scan...\n",
    "meta": {},
    "output": {
      "scan_id": "",
      "was_scan_skipped": true
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/urlscan:2.3.0 --debug run < tests/submit_url_for_scan.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "rapid7/urlscan.io:2.3.0. Step name: submit_url_for_scan\n",
    "meta": {},
    "output": {
      "scan_id": "044e0dfc-bea8-4a41-9c80-dd3275571ec1",
      "was_scan_skipped": false
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/urlscan:2.3.0 --debug run < tests/submit_url_for_scan2.json
</summary>
</details>

<details>

```
[*] Use ``make menu`` for available targets
[*] Including available Makefiles: ../tools/Makefiles/Helpers.mk ../tools/Makefiles/Colors.mk
--
[*] Running validators
[*] Validating plugin at .

[*] Running Integration Validators...
[*] Executing validator HelpValidator
[*] Executing validator ChangelogValidator
[*] Executing validator CloudReadyConnectionCredentialTokenValidator
[*] Executing validator RequiredKeysValidator
[*] Executing validator UseCaseValidator
[*] Executing validator SpecPropertiesValidator
[*] Executing validator SpecVersionValidator
[*] Executing validator FilesValidator
[*] Executing validator TagValidator
[*] Executing validator DescriptionValidator
[*] Executing validator TitleValidator
[*] Executing validator VendorValidator
[*] Executing validator DefaultValueValidator
[*] Executing validator IconValidator
[*] Executing validator RequiredValidator
[*] Executing validator VersionValidator
[*] Executing validator DockerfileParentValidator
[*] Executing validator LoggingValidator
[*] Executing validator ProfanityValidator
[*] Executing validator AcronymValidator
[*] Executing validator JSONValidator
[*] Executing validator OutputValidator
[*] Executing validator RegenerationValidator
[*] Executing validator HelpInputOutputValidator
[*] Executing validator SupportValidator
[*] Executing validator RuntimeValidator
[*] Executing validator VersionPinValidator
[*] Executing validator EncodingValidator
[*] Executing validator ExampleInputValidator
[*] Executing validator CloudReadyValidator
[*] Plugin successfully validated!

----
[*] Total time elapsed: 1662.819ms

[*] Validating spec with js-yaml
[SUCCESS] Passes js-yaml spec check


[*] Validating markdown...
[SUCCESS] Passes markdown linting

[*] Validating python files for style...
[SUCCESS] Passes flake8 linting

[*] Validating python files for security vulnerabilities...
[main]  INFO    profile include tests: None
[main]  INFO    profile exclude tests: None
[main]  INFO    cli include tests: None
[main]  INFO    cli exclude tests: None
[main]  INFO    running on Python 3.7.3
Run started:2021-04-25 15:05:49.617740

Test results:
        No issues identified.

Code scanned:
        Total lines of code: 842
        Total lines skipped (#nosec): 0

Run metrics:
        Total issues (by severity):
                Undefined: 0.0
                Low: 0.0
                Medium: 0.0
                High: 0.0
        Total issues (by confidence):
                Undefined: 0.0
                Low: 0.0
                Medium: 0.0
                High: 0.0
Files skipped (0):
[SUCCESS] Passes bandit security checks

[*] Checking for typos with Misspell...
[SUCCESS] Passes Misspell check
```

<summary>
make validate
</summary>
</details>

<details>

```
[*] Validating plugin with all validators at .

[*] Running Integration Validators...
[*] Executing validator HelpValidator
[*] Executing validator ChangelogValidator
[*] Executing validator CloudReadyConnectionCredentialTokenValidator
[*] Executing validator RequiredKeysValidator
[*] Executing validator UseCaseValidator
[*] Executing validator SpecPropertiesValidator
[*] Executing validator SpecVersionValidator
[*] Executing validator FilesValidator
[*] Executing validator TagValidator
[*] Executing validator DescriptionValidator
[*] Executing validator TitleValidator
[*] Executing validator VendorValidator
[*] Executing validator DefaultValueValidator
[*] Executing validator IconValidator
[*] Executing validator RequiredValidator
[*] Executing validator VersionValidator
[*] Executing validator DockerfileParentValidator
[*] Executing validator LoggingValidator
[*] Executing validator ProfanityValidator
[*] Executing validator AcronymValidator
[*] Executing validator JSONValidator
[*] Executing validator OutputValidator
[*] Executing validator RegenerationValidator
[*] Executing validator HelpInputOutputValidator
[*] Executing validator SupportValidator
[*] Executing validator RuntimeValidator
[*] Executing validator VersionPinValidator
[*] Executing validator EncodingValidator
[*] Executing validator ExampleInputValidator
[*] Executing validator CloudReadyValidator
[*] Executing validator ExceptionValidator
[*] Executing validator CredentialsValidator
[*] Executing validator PasswordValidator
[*] Executing validator PrintValidator
[*] Executing validator ConfidentialValidator
[*] Executing validator DockerValidator
[*] Executing validator URLValidator
WARNING: URLs found that return a 4xx code. Verify they are publicly accessible and if not, update with a working URL.
violation: help.md[286]: https://s.mapleleafjobs.net/f/a/1_Qyh4s9oyxnFuau29s_HA~~/AAF10QA~/RgRgsQBWP0Q2aHR0cHM6Ly9yZC5tYXBsZWxlYWZqb2JzLm5ldC9hLz94PVQxVVc5YmFtbDhiLTMzajZiLTIxVwNzcGNCCgAoVnvOXtJ8SVxSHWFtYmVyLm1hbGxldHRAcGFjaWZpY2xpZmUuY29tWAQAAABM
violation: help.md[288]: https://s.mapleleafjobs.net/f/a/1_Qyh4s9oyxnFuau29s_HA~~/AAF10QA~/RgRgsQBWP0Q2aHR0cHM6Ly9yZC5tYXBsZWxlYWZqb2JzLm5ldC9hLz94PVQxVVc5YmFtbDhiLTMzajZiLTIxVwNzcGNCCgAoVnvOXtJ8SVxSHWFtYmVyLm1hbGxldHRAcGFjaWZpY2xpZmUuY29tWAQAAABM
violation: help.md[463]: https://s.mapleleafjobs.net/f/a/1_Qyh4s9oyxnFuau29s_HA~~/AAF10QA~/RgRgsQBWP0Q2aHR0cHM6Ly9yZC5tYXBsZWxlYWZqb2JzLm5ldC9hLz94PVQxVVc5YmFtbDhiLTMzajZiLTIxVwNzcGNCCgAoVnvOXtJ8SVxSHWFtYmVyLm1hbGxldHRAcGFjaWZpY2xpZmUuY29tWAQAAABM
[*] Plugin successfully validated!

----
[*] Total time elapsed: 28111.912ms
```

<summary>
icon-validate --all . 
</summary>
</details>

### CODE REVIEW

**Comment by mrinehart-r7** [comment]
@r7-kszczepanskagorna I'm not sure this is the right solution to what was in the ticket... It will be more apparent in the logs, but it may make this harder to use within a workflow since now a workflow developer will need to check whether or not the Scan ID output is empty. Can we add a boolean output to this as well - maybe something like "Was Scan Skipped" - so that all a developer needs to do is check if that boolean is true/false?

**Comment by r7-kszczepanskagorna** [comment]
@mrinehart-r7 Sure :) I will take care of this :)

**Comment by jschipp-r7** [comment]
@r7-kszczepanskagorna Bump :)

File: urlscan/help.md [hunk 1]
```diff
@@ -821,6 +823,7 @@ _This plugin does not contain any troubleshooting information._
 
 # Version History
 
+* 2.3.0 - In action Submit URL for Scan add logger when submitted domain in on blacklist
```
Comment (mrinehart-r7) [inline urlscan/help.md:826]:
```suggestion
* 2.3.0 - Add logger when submitted domain is in blacklist for Submit URL for Scan action
```

**Review by mrinehart-r7 — CHANGES_REQUESTED** [review]

**Review by mrinehart-r7 — DISMISSED** [review]

**Review by jmcadams-r7 — DISMISSED** [review]

**Review by jmcadams-r7 — APPROVED** [review]

**Review by mrinehart-r7 — APPROVED** [review]

**Pull request merged**